syntax = "proto3";
package instance;




import "google/protobuf/struct.proto";

/*
SearchTotal请求
*/
message SearchTotalRequest {
    /*
    模型对象ID
    */
    string objectId = 1 ;
    /*
    查询条件, 与[实例搜索接口]的query字段说明一致
    */
    google.protobuf.Struct query = 2 ;
    /*
    按照权限过滤[权限列表]
    */
    repeated string permission = 3 ;
    /*
    是否只获取与自己有关的那部分数据, 默认为false
    */
    bool only_my_instance = 4 ;
}

/*
SearchTotal返回
*/
message SearchTotalResponse {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    错误信息
    */
    string error = 2 ;
    /*
    返回消息
    */
    string message = 3 ;
    /*
    实例总数
    */
    int32 data = 4 ;
}

/*
SearchTotalApi返回
*/
message SearchTotalResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    instance.SearchTotalResponse data = 4 ;
}

